src/main/scala/mesosphere/mesos/Availability.scala

```diff
@@ -0,0 +1,23 @@
+package mesosphere.mesos
+
+import org.apache.mesos.Protos.Offer
+
+object Availability {
+
+  def offerAvailable(offer: Offer): Boolean = {
+    var unavailability = offer.hasUnavailability
+    var now = System.nanoTime()
+    if (offer.hasUnavailability && offer.getUnavailability.hasStart) {
+      val start = offer.getUnavailability.getStart.getNanoseconds
+      if (now.>=(start)) {
+        if (offer.getUnavailability.hasDuration) {
+          return start.+(offer.getUnavailability.getDuration.getNanoseconds).<(now)
```

Comment:
@unterstein what you are saying is to implement maintenance primitives of mesos. And when a mode is in maintenence in mesos, mesos will keep offering its resources to frameworks, and its up to framework to accept the offers or not. 

 If you follow the https://jira.mesosphere.com/browse/MARATHON-3216 comments. It was already supposed to divide the task into two steps. 

The first step you are saying. This PR is already implementing only that. Dropping offers in maintenance mode. 
And the second step when you say implement a shift, it means implement inverse offers. And marathon needs to shift to v1 API of mesos to be able to do that.